### ISSUE TITLE
Server-Name-Indication  for MQTT connections

### ISSUE THREAD

**Issue by mworrell** [thomasnordquist/MQTT-Explorer#130]
We are currently adding MQTT listeners to https://github.com/zotonic/zotonic and using MQTT-Explorer for manual testing and inspection.

Zotonic uses vhosts with a MQTT broker per site.

In the `ws` connection Server-Name-Indication is used.
Though it seems that in the `mqtt` connection type Server-Name-Indication is not used.

Could Server-Name-Indication be added to the `mqtt` connection type?

Thanks for this great utility.

**Comment by thomasnordquist** [comment]
I'm glad you like it 😄

Added support with https://github.com/thomasnordquist/MQTT-Explorer/commit/261454986e3ff66592aa183407e1fb68d1ec9c65 (assuming the Node TLS documentation is correct)

I also prepared a pre-release for you to test the changes:
https://github.com/thomasnordquist/MQTT-Explorer/releases/tag/v0.0.0-0.3.1-alpha-sni

Would be awesome if you tell me if it works!

**Comment by thomasnordquist** [comment]
In the current implementation, it appears the port number is part of the hostname.
Just started working on client certificate support when I noticed.

**Comment by thomasnordquist** [comment]
Check out https://github.com/thomasnordquist/MQTT-Explorer/releases/tag/v0.0.0-0.3.1-alpha-sni

**Comment by mworrell** [comment]
You are fast!  I checked the release candidate and it works like a charm.

```
09:38:46.918 [info] DEBUG: z_ssl_certs:63  {<0.31085.2>,"t.lamma"}
```

Hostname comes through, exactly how it is expected in the SNI routines.

**Comment by thomasnordquist** [comment]
Glad to hear it, will be part of an official release soon.

**Comment by thomasnordquist** [comment]
Released with [v0.3.5](https://github.com/thomasnordquist/MQTT-Explorer/releases/tag/v0.3.5).

**Comment by mworrell** [comment]
Thanks!
